gef.py

```diff
@@ -10482,21 +10482,83 @@ def lookup_command(self, cmd):
 
         return None
 
+@register_command
+class AliasesCommand(GenericCommand):
+    """Base command to add, remove, or list aliases."""
 
-class GefAliases(gdb.Command):
-    """List all custom aliases."""
+    _cmdline_ = "aliases"
+    _syntax_  = "{:s} (add|rm|ls)".format(_cmdline_)
 
     def __init__(self):
-        super().__init__("aliases", gdb.COMMAND_OBSCURE, gdb.COMPLETE_NONE)
+        super().__init__(prefix=True)
         return
 
-    def invoke(self, args, from_tty):
-        self.dont_repeat()
-        ok("Aliases defined:")
-        for _alias in __aliases__:
-            gef_print("{:30s} {} {}".format(_alias._alias, RIGHT_ARROW, _alias._command))
+    def do_invoke(self, argv):
+        self.usage()
+        return
+
+@register_command
+class AliasesAddCommand(AliasesCommand):
+    """Command to add aliases."""
+
+    _cmdline_ = "aliases add"
+    _syntax_  = "{0} [ALIAS] [COMMAND]".format(_cmdline_)
+    _example_ = "{0} scope telescope".format(_cmdline_)
+
+    def __init__(self):
+        super().__init__()
+        return
+
+    def do_invoke(self, argv):
+        if (len(argv) < 2):
+            self.usage()
+            return
+        GefAlias(argv[0], " ".join(argv[1:]))
+        GefSaveCommand().invoke(None, False)
+        return
+
+@register_command
+class AliasesRmCommand(AliasesCommand):
+    """Command to remove aliases."""
+
+    _cmdline_ = "aliases rm"
+    _syntax_ = "{0} [ALIAS]".format(_cmdline_)
+
+    def __init__(self):
+        super().__init__()
+        return
+
+    def do_invoke(self, argv):
+        global __aliases__
+        if len(argv) != 1:
+            self.rm_usage()
+            return
+        try:
+            alias_to_remove = next(filter(lambda x: x._alias == argv[0], __aliases__))
+            __aliases__.remove(alias_to_remove)
+        except (ValueError, StopIteration) as e:
+            err("{0} not found in aliases.".format(argv[0]))
+            return
+        GefSaveCommand().invoke(None, False)
```

Comment:
Same question here.

If not necessary for your script (I don't see why it would be, but I may be wrong), it'd be better to simply replace the call to `GefSaveCommand().invoke()` with a message like 
```
added alias `foo` to the session, enter 'gef save' to make it permanent
```

Reply:
I think this message ^ is helpful, both for adding and removing aliases. @hoefler02 can you add them?